test/tests/functional/pbs_sched_rerun.py

```diff
@@ -0,0 +1,119 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2021 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of both the OpenPBS software ("OpenPBS")
+# and the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# OpenPBS is free software. You can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+#
+# OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# PBS Pro is commercially licensed software that shares a common core with
+# the OpenPBS software.  For a copy of the commercial license terms and
+# conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+# Altair Legal Department.
+#
+# Altair's dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of OpenPBS and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair's trademarks, including but not limited to "PBS™",
+# "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+# subject to Altair's trademark licensing policies.
+
+
+from tests.functional import *
+
+
+hook_body = """
+import pbs
+e = pbs.event()
+if e.type == pbs.EXECJOB_BEGIN:
+    e.job.resources_used["foo"] = 123
+    if e.job.run_count == 1:
+        for v in e.vnode_list:
+            if (v == "%s"):
```

Comment:
You don't need the ().  Also, will this ever be true?  "%s" is a string of exactly that.  The vnode names won't be "%s".  It'll be the name of the vnode.

Reply:
Yes. As you can see in the ptl log, the value of the imported hook is:
```
2024-03-14 13:34:32,681 INFOCLI  rocky9(run_cmd): sudo -H /opt/pbs/bin/qmgr -c import hook rerun_job application/x-python default /tmp/PtlPbso6ye7_s9
2024-03-14 13:34:32,759 INFO     server rocky9: imported hook body
---

import pbs
e = pbs.event()
if e.type == pbs.EXECJOB_BEGIN:
	e.job.resources_used["foo"] = 123
	if e.job.run_count == 1:
		for v in e.vnode_list:
			if (v == "rocky9"):
				e.vnode_list[v].state = pbs.ND_OFFLINE
		e.job.rerun()
		e.reject("reruen job")
e.accept()
---
2024-03-14 13:34:32,760 INFOCLI2 rocky9(run_cmd): sudo -H /opt/pbs/bin/pbsnodes -a -Fjson
```
This is intentional and I want to set offline the `self.hostA`.
Of course, I will remove the parenthesis. Thank you.

Reply:
Oh I see
I didn't notice where you did the substitution down when you created the hook.

Since you only need that hook in one test, you could put the variable in the test itself and use an f""" string with {self.hostA}  instead of using the substitution.  That way it'll be clearer of what it's being replaced with.